javascript/packages/fury/lib/classResolver.ts

```diff
@@ -94,30 +94,30 @@ export default class SerializerResolver {
     return this.customSerializer[tag];
   }
 
-  tagToBuffer(tag: string) {
-    const tagBuffer = fromUint8Array(new TextEncoder().encode(tag));
-    let tagHash = x64hash128(tagBuffer, 47).getBigUint64(0);
-    if (tagHash === BigInt(0)) {
-      tagHash = BigInt(1);
-    }
+  createTagWriter(tag: string) {
+    let tagIndex = -1;
```

Comment:
Each tag call createTagWriter one time, so the tagIndex should be reset before start serialize. Perhaps we can store 
 the tagIndex on classResolver, and reset it to zero when start serialize.
The code maybe like this:
```javascript
 this.writeStringIndex = [];
 reset() {
    this.writeStringIndex.fill(-1);
 }
 createTagWriter(tag: string) {
    this.writeStringIndex.push(-1);
    let idx = this.writeStringIndex.length - 1;
    return () =>  {
        const tagIndex = this.writeStringIndex[idx];
       // when update
       this.writeStringIndex[idx] = this.writeStringCount++;
    }
}
```

Reply:
done